app/src/org/commcare/dalvik/activities/FormRecordListActivity.java

```diff
@@ -177,17 +173,18 @@ public void onCreate(Bundle savedInstanceState) {
                 filterSelect.setAdapter(spinneritems);
                 spinneritems.setDropDownViewResource(R.layout.form_filter_item);
                 filterSelect.setOnItemSelectedListener(new OnItemSelectedListener() {
-                    
                     /*
                      * (non-Javadoc)
                      * @see android.widget.AdapterView.OnItemSelectedListener#onItemSelected(android.widget.AdapterView, android.view.View, int, long)
                      */
                     @Override
                     public void onItemSelected(AdapterView<?> arg0, View arg1, int index, long id) {
-                        adapter.setFormFilter(FormRecordFilter.values()[index]);
-                        adapter.resetRecords();
-                        adapter.notifyDataSetChanged();
-                        
+                        // NOTE: This gets called every time a spinner gets
+                        // set-up and also every time spinner state is restored
+                        // on scree-rotation. Hence we defer onCreate record
+                        // loading until this gets triggered automatically.
+                        adapter.setFilterAndResetRecords(FormRecordFilter.values()[index]);
+
```

Comment:
onItemSelected gets called every time this activity is setup/rotated. Use this fact to launch record loading. It might get called multiple times if restoring a previous selection, so I've included logic in the method making sure record reloading only occurs when the filter actually changes.